apps/test/unit/storage/dataUtilsTest.js

```diff
@@ -25,6 +26,26 @@ describe('isBlank', () => {
   });
 });
 
+describe('sortAlphabeticallyOrNumerically', () => {
+  expect(sortAlphabeticallyOrNumerically([])).to.deep.equal([]);
+  expect(sortAlphabeticallyOrNumerically([10, 1, 4, 12])).to.deep.equal([
+    1,
+    4,
+    10,
+    12
+  ]);
+  expect(
+    sortAlphabeticallyOrNumerically([
+      'abc',
+      'red',
+      'blue',
+      'green',
+      'words',
+      'CAPS'
+    ])
+  ).to.deep.equal(['CAPS', 'abc', 'blue', 'green', 'red', 'words']);
```

Comment:
This is an interesting result.  Is this behavior actually the user experience we want in the dropdown? If you'd prefer a natural-sort, then [`String.prototype.localeCompare`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare) may solve your problem.  We use this several places in our project already.

https://github.com/code-dot-org/code-dot-org/blob/1cad5ac667f9ed1a3aa31063e6d06bd21a78e2bd/apps/src/netsim/NetSimLogBrowserFilters.jsx#L158-L159

There are also some notes on [this StackOverflow post](https://stackoverflow.com/questions/2802341/javascript-natural-sort-of-alphanumerical-strings) about doing this efficiently on large sets.